### PR TITLE
Hide entry.js | Add setup command to restore

### PR DESCRIPTION
**Pull request by dac09**
### What?
Hide's entry.js (the mount point of the web side), and houses it internally within `@redwoodjs/web`

If you're a power user 💪🏽, and want to customise it, simply run `yarn rw setup custom-entry` and it'll add it back to your project. 

### Help needed 🚩
- Suggest a better name for the setup function
- Is there any code I can add to make it easier for the typescript transition?

### Note
⚠ This is a breaking change, whether you're opt-ing in to prerender or not.

### CODE REVIEW

**Comment by peterp** [comment]
Looking good! 👍

File: packages/cli/src/commands/setup/custom-entry/custom-entry.js
Comment (peterp) [inline packages/cli/src/commands/setup/custom-entry/custom-entry.js:31]:
```suggestion
        const entryJsFile = path.join(getPaths().web.src, 'entry.js')
```

Comment (peterp) [reply packages/cli/src/commands/setup/custom-entry/custom-entry.js:31]: Do we have this mapped to `getPaths()`?

File: packages/cli/src/commands/setup/custom-entry/custom-entry.js
Comment (peterp) [inline packages/cli/src/commands/setup/custom-entry/custom-entry.js:40]: I usually use `paths.join(getPaths().base, 'node_modules/@redwoodjs/web/entry/index.js)` so that if this file moves around that it's not going to cause problems.

**Review by peterp — CHANGES_REQUESTED** [review]
A few small things that I noticed 👍

File: packages/core/config/webpack.common.js
Comment (peterp) [inline packages/core/config/webpack.common.js:178]: `path.join()` might be easier to understand.

File: packages/cli/src/commands/setup/custom-entry/custom-entry.js
Comment (dac09) [reply packages/cli/src/commands/setup/custom-entry/custom-entry.js:40]: hmm sure makes sense

File: packages/cli/src/commands/setup/custom-entry/custom-entry.js
Comment (dac09) [reply packages/cli/src/commands/setup/custom-entry/custom-entry.js:31]: you're right I did add it, forgot about it

File: packages/cli/src/commands/setup/custom-entry/custom-entry.js
Comment (dac09) [inline packages/cli/src/commands/setup/custom-entry/custom-entry.js:60]: Needs writing up first :)

Comment (peterp) [reply packages/cli/src/commands/setup/custom-entry/custom-entry.js:60]: This is looking great, approved, pending those docs :)

**Review by peterp — APPROVED** [review]
This is looking great, approved, pending those docs :)

**Comment by thedavidprice** [comment]
Taking this for an E2E spin now. Specifically double-checking the template codebase changes (for my sanity).

**Comment by thedavidprice** [comment]
@dac09 CI lint error. Ignore?

Screen Shot 2021-02-24 at 9 54 51 AM

**Comment by dac09** [comment]
> @dac09 CI lint error. Ignore?

Yeah must be the new eslint configs that just went in. I'll see if I can selectively ignore these when I update this branch

**Review by thedavidprice — CHANGES_REQUESTED** [review]
Result of E2E test is a runtime error that's blocking merge:
Screen Shot 2021-02-24 at 11 03 37 AM

Looks like you get a Babel ChooseYourOwnAdventure™
```bash
Add @babel/preset-react (https://git.io/JfeDR) to the 'presets' section of your Babel config to enable transformation.

If you want to leave it as-is, add @babel/plugin-syntax-jsx (https://git.io/vb4yA) to the 'plugins' section to enable parsing.
```

**Comment by dac09** [comment]
Can confirm :(. It works ok when building, but no when running `yarn rw dev` unfortunately, hence the e2e test failing I think. So glad you ran that 👍

**Comment by dac09** [comment]
Ok @thedavidprice, e2e tests now pass for me. Sorry about this. Please give it a go on your side!

I'm really sad I couldn't use my elegant template solution for the setup command 😭

**Review by thedavidprice — APPROVED** [review]

File: packages/cli/src/commands/setup/custom-entry/custom-entry.js
Comment (jtoar) [reply packages/cli/src/commands/setup/custom-entry/custom-entry.js]: working on those here https://github.com/redwoodjs/redwoodjs.com/pull/589

**Pull request merged**
